test/testvideocapture.c

```diff
@@ -0,0 +1,787 @@
+
+    int start = 1;
+    SDL_VideoCaptureFrame frame_current;
+
+#if USE_DMABUF
+    EGLDisplay display;
+
+    typedef EGLDisplay (*eglGetCurrentDisplay_t)();
+    typedef EGLImageKHR (*eglCreateImageKHR_t)(EGLDisplay dpy, EGLContext ctx, EGLenum target, EGLClientBuffer buffer, const EGLint *attrib_list);
+    typedef EGLBoolean (*eglDestroyImageKHR_t)(EGLDisplay dpy, EGLImageKHR image);
+    typedef void (*glActiveTexture_t)(GLenum texture);
+    typedef void (*glEGLImageTargetTexture2DOES_t)(GLenum target, GLeglImageOES image);
+    typedef EGLint (*eglGetError_t)(void);
+
+    eglGetCurrentDisplay_t eglGetCurrentDisplay;
+    eglCreateImageKHR_t eglCreateImageKHR;
+    eglDestroyImageKHR_t eglDestroyImageKHR;
+    glActiveTexture_t glActiveTexture;
+    glEGLImageTargetTexture2DOES_t glEGLImageTargetTexture2DOES;
+    eglGetError_t eglGetError;
+
+#endif
+
+    measure_fps_t fps_main;
+    measure_fps_t fps_capture;
+
+    SDL_zero(fps_main);
+    SDL_zero(fps_capture);
+    SDL_zero(frame_current);
+
+    SDL_Log("%s", usage);
+
+    /* Enable standard application logging */
+    SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
+
+#if USE_DMABUF
+    SDL_SetHint("SDL_RENDER_OPENGL_NV12_RG_SHADER", "1");
+    SDL_SetHint("SDL_VIDEO_X11_FORCE_EGL", "1"); /* Don't use GLX */
+    SDL_SetHint(SDL_HINT_RENDER_DRIVER, "opengles2");
+#endif
+
+    /* Load the SDL library */
+    if (SDL_Init(SDL_INIT_VIDEO) < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't initialize SDL: %s", SDL_GetError());
+        return 1;
+    }
+
+    if (!(window = SDL_CreateWindow("Local Video", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1000, 800, 0))) {
+        SDL_Log("Couldn't create window: %s", SDL_GetError());
+        return 1;
+    }
+
+    SDL_LogSetAllPriority(SDL_LOG_PRIORITY_VERBOSE);
+
+    renderer = SDL_CreateRenderer(window, NULL, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);
+    if (renderer == NULL) {
+        SDL_Log("Couldn't create renderer: %s", SDL_GetError());
+        return 1;
+    }
+
+#if USE_DMABUF
+
+#define LOAL_FUNC(f)                            \
+    f = (f ##_t)SDL_GL_GetProcAddress(#f);      \
+    if (f == NULL) {                            \
+        SDL_Log("Cannot load function: " #f );  \
+        return 0;                               \
+    }                                           \
+
+    LOAL_FUNC(eglGetError)
+    LOAL_FUNC(eglGetCurrentDisplay)
+    LOAL_FUNC(eglCreateImageKHR)
+    LOAL_FUNC(eglDestroyImageKHR)
+    LOAL_FUNC(glActiveTexture)
+    LOAL_FUNC(glEGLImageTargetTexture2DOES)
+
+    display = eglGetCurrentDisplay();
+    if (display == EGL_NO_DISPLAY) {
+        SDL_Log("no dispay");
+        return 0;
+    }
+#endif
+
+
+    SDL_LogSetAllPriority(SDL_LOG_PRIORITY_VERBOSE);
+
+    id = SDL_OpenVideoCapture(NULL);
+
+    if (id == SDL_VIDEO_CAPTURE_INVALID) {
+        SDL_Log("Error SDL_OpenVideoCapture: %s", SDL_GetError());
```

Comment:
Is it okay to continue execution if the video capture could not be opened?

Reply:
yes, maybe. app could wait that you plug a usb camera